reactive/kotlinx-coroutines-rx2/test/ObservableTest.kt

```diff
@@ -137,28 +134,4 @@ class ObservableTest : TestBase() {
             expect(4)
         }
     }
-
-    @Test
-    fun testExceptionAfterCancellation() {
-        // Test that no exceptions were reported to the global EH (it will fail the test if so)
-        val handler = { e: Throwable ->
-            assertFalse(e is CancellationException)
-        }
-        withExceptionHandler(handler) {
-            RxJavaPlugins.setErrorHandler {
-                require(it !is CancellationException)
-            }
-            Observable
-                .interval(1, TimeUnit.MILLISECONDS)
-                .take(1000)
-                .switchMapSingle {
-                    rxSingle {
-                        timeBomb().await()
-                    }
-                }
-                .blockingSubscribe({}, {})
-        }
-    }
```

Comment:
This test seems to have been taken from the issue reported at #252. However it's complicated and not well implemented (I originally tried to improve it in commit 58ad0cdfca81ab2d5c5c410f889293c51beede75). Also, it was actually testing `rxSingle` instead of `rxObservable`, which is why it didn't catch the issue in the first place that this pull request is trying to solve. `rxSingle` (and the other RX builders) already have a simpler and deterministic test for undeliverable error handling (`testUnhandledException()`), which I have copied over in the `ObservableExceptionHandlingTest` test suite, so I deleted this test since should be no further need for it.